examples/list-streams.js

```diff
@@ -0,0 +1,188 @@
+/*
+    This example demonstrates how to get streams from given facility and get stream details (name, parent). It uses 2-legged
+    authentication - this requires that application is added to facility as service.
+*/
+async function main() {
+    // STEP 1 - obtain token. The sample uses 2-legged token but it would also work with 3-legged token
+    // assuming that user has access to the facility
+    const token = await createToken('YOUR_CLIENT_ID',
+        'YOUR_CLIENT_SECRET', 'data:read');
+    
+    // STEP 2 - get facility and default model. The default model has same id as facility
+    const facilityId = 'YOUR_FACILITY_URN';
+    const facility = await getFacility(token, facilityId);
+    const defaultModelId = facilityId.replace('urn:adsk.dtt:', 'urn:adsk.dtm:');
+    const defaultModel = facility.links.find((m) => {
+        return  m.modelId === defaultModelId;
+    });
+
+    // STEP 3 - get streams and print stream + its parent
+    const streams = await getStreams(token, defaultModel.modelId);
+
+    for (const stream of streams) {
+        // host is stored as parent
+        const parentXref = stream['x:p'];
+
+        if (!parentXref) {
+            continue;
+        }
+        // the id of the host is encoded
+        const [ modelId, key ] = decodeXref(parentXref);
+
+        const elementData = await getElementData(token, `urn:adsk.dtm:${modelId}`, key);
```

Comment:
I think, we want to collect parents here and make a single batch request to get their properties. it is more in line with general Tandem API approach - most of our core APIs support batch operations.

Reply:
Refactored code to use batch query.